scripts/suite_objects.py

```diff
@@ -2213,14 +2378,39 @@ def write(self, outfile, host_arglist, indent, const_mod,
             self.run_env.logger.debug(msg.format(self.name, call_vars))
         # end if
         self.call_list.declare_variables(outfile, indent+1, dummy=True)
-        if subpart_vars:
+        # DECLARE local variables
+        if subpart_allocate_vars:
```

Comment:
I think this should technically be:

```suggestion
        if subpart_allocate_vars or subpart_scalar_vars or subpart_optional_vars:
```

Reply:
True. Fixed.